client/network/src/protocol/notifications/behaviour.rs

```diff
@@ -955,7 +950,24 @@ impl Notifications {
 
 		match mem::replace(state, PeerState::Poisoned) {
 			// Incoming => Disabled
-			PeerState::Incoming { mut connections, backoff_until } => {
+			PeerState::Incoming { mut connections, backoff_until, incoming_index } => {
+				if index < incoming_index {
```

Comment:
Why is this change still relevant? I understood it was part of making the channel asynchronous but if it's back to synchronous again, this is not needed or am I missing something?

Reply:
The communication is asynchronous both directions via the unbounded channels. The only part that is synchronous is polling of `PeerSet` stub, which doesn't really change anything.